ra/ra.go

```diff
@@ -1940,8 +1953,10 @@ func (ra *RegistrationAuthorityImpl) updateRevocationForKeyCompromise(ctx contex
 	_, err = ra.SA.UpdateRevokedCertificate(ctx, &sapb.RevokeCertificateRequest{
 		Serial:     serialString,
 		Reason:     int64(ocsp.KeyCompromise),
-		DateNS:     thisUpdate,
+		DateNS:     now.UnixNano(),
+		Date:       timestamppb.New(now),
 		BackdateNS: status.RevokedDateNS,
+		Backdate:   status.RevokedDate,
```

Comment:
This is reading a field.

Reply:
I assume that `Date: timestamppb.New(now)` 2 lines up should also be removed?

Reply:
No, that one's good. The issue isn't that we're populating the `Date` and `Backdate` fields -- that's good. The issue is that this line is reading _from_ `status.RevokedDate`, which isn't guaranteed to exist yet: `status` is a `corepb.CertificateStatus` message retrieved from the SA, which might not yet be populating the `Status` field of that message depending on deployment order.

The `Date: timestamppb.New(now)` is fine, because we know that `now` exists (we made it ourselves).

This line needs to be updated to be something like `Backdate: timestamppb.New(time.Unix(0, status.RevokedDateNS))`, so that we're only reading _from_ the old `RevokedDateNS` field.

Reply:
That makes sense. I've found a 2nd location in `//ra/ra.go` where I could call `timestamppb.New(time.Unix(0, status.RevokedDateNS))`